PartitionMigrationListenerTest.testMigrationStats_afterPartitionsLost_when_NO_MIGRATION

http://jenkins.hazelcast.com/job/Hazelcast-pr-builder/3601/

```
Expected: a value equal to or greater than <4>
     but: <3> was less than <4>

Stacktrace

java.lang.AssertionError: 

Expected: a value equal to or greater than <4>
     but: <3> was less than <4>
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at org.junit.Assert.assertThat(Assert.java:956)
	at org.junit.Assert.assertThat(Assert.java:923)
	at com.hazelcast.partition.PartitionMigrationListenerTest$EventCollectingMigrationListener.lambda$awaitEventPacksComplete$0(PartitionMigrationListenerTest.java:445)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1308)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1410)
	at com.hazelcast.partition.PartitionMigrationListenerTest$EventCollectingMigrationListener.awaitEventPacksComplete(PartitionMigrationListenerTest.java:444)
	at com.hazelcast.partition.PartitionMigrationListenerTest$EventCollectingMigrationListener.ensureAndGetEventPacks(PartitionMigrationListenerTest.java:435)
	at com.hazelcast.partition.PartitionMigrationListenerTest.testMigrationStats_afterPartitionsLost_when_NO_MIGRATION(PartitionMigrationListenerTest.java:178)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:114)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:106)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```

--- a/hazelcast/src/test/java/com/hazelcast/partition/PartitionMigrationListenerTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/partition/PartitionMigrationListenerTest.java
@@ -160,7 +160,9 @@ public class PartitionMigrationListenerTest extends HazelcastTestSupport {
     @Test
     public void testMigrationStats_afterPartitionsLost_when_NO_MIGRATION() {
         TestHazelcastInstanceFactory factory = createHazelcastInstanceFactory();
-        HazelcastInstance[] instances = factory.newInstances(new Config(), 10);
+        Config config = new Config().setProperty(GroupProperty.PARTITION_COUNT.getName(), "2000");
+        HazelcastInstance[] instances = factory.newInstances(config, 10);
+        assertClusterSizeEventually(instances.length, instances);
         warmUpPartitions(instances);
 
         EventCollectingMigrationListener listener = new EventCollectingMigrationListener();
@@ -222,22 +224,7 @@ public class PartitionMigrationListenerTest extends HazelcastTestSupport {
 
         assertNotNull(lastProgress);
         assertEquals(migrationResult.getCompletedMigrations(), lastProgress.getCompletedMigrations());
-//        assertEquals(migrationResult.getTotalElapsedTime(), lastProgress.getTotalElapsedTime());
         assertThat(migrationResult.getTotalElapsedTime(), greaterThanOrEqualTo(lastProgress.getTotalElapsedTime()));
-    }
-
-    private void assertPromotionEventsConsistentWithResult(MigrationEventsPack eventsPack) {
-        List<ReplicaMigrationEvent> completedMigrations = eventsPack.migrationsCompleted;
-        MigrationState migrationResult = eventsPack.migrationProcessCompleted;
-        assertEquals(migrationResult.getCompletedMigrations(), completedMigrations.size());
-
-        for (ReplicaMigrationEvent event : completedMigrations) {
-            assertTrue(event.isSuccess());
-            MigrationState progress = event.getMigrationState();
-            assertEquals(migrationResult.getStartTime(), progress.getStartTime());
-            assertEquals(migrationResult.getPlannedMigrations(), progress.getPlannedMigrations());
-            assertEquals(migrationResult.getCompletedMigrations(), progress.getCompletedMigrations());
-        }
     }
 
     @Test

